src/workerd/util/sqlite.c++

```diff
@@ -174,16 +181,24 @@ class SqliteCallScope {
     /* SQLITE_MISUSE doesn't put error info on the database object, so check it separately */      \
     KJ_ASSERT(_ec != SQLITE_MISUSE, "SQLite misused: " #code, ##__VA_ARGS__);                      \
     if (_ec == SQLITE_IOERR) sqliteCallScope.rethrowVfsError();                                    \
+    if (isCriticalError(code)) {                                                                   \
```

Comment:
I would move this check into `SQLITE_REQUIRE` rather than reproduce it in both variants of `SQLITE_CALL`.